OpenJS-SBOM-CSCRM-Challenges-Recommendations.md

```diff
@@ -0,0 +1,338 @@
+   * Despite the presence of ecosystem native  package-lock.json and npm-shrinkwrap.json manifest files this information is more easily consumed by end users in an cross-ecosystem standard format.   
+8. **Software Supply Chain Monitoring Services:** When consuming SBOMs to determine the potential scope of vulnerabilities across products, these services should limit their intake to exclusively consuming JavaScript Application SBOMs.
+
+# **Attestation and Package Provenance**
+
+
+
+### **Attestation Definition**
+
+
+According to OpenSSF’s [SLSA Project](https://slsa.dev/attestation-model), software attestation is “*an authenticated statement (metadata) about a software artifact or collection of software artifacts.*”
+
+Software attestation comes in many forms, including:
+
+* Application Code Signing (eg: [Apple](https://developer.apple.com/documentation/security/code-signing-services), [Google](https://developer.android.com/studio/publish/app-signing), [Microsoft](https://learn.microsoft.com/en-us/windows/security/application-security/application-control/app-control-for-business/deployment/use-code-signing-for-better-control-and-protection))  
+* Attestations of Containers and their SBOMs (eg: [Open Container Initiative](https://opencontainers.org/community/certified/))  
+* Package Provenance (eg: [npm](https://github.blog/security/supply-chain-security/introducing-npm-package-provenance/), [pypi](https://github.com/trailofbits/pypi-attestations))
+
+### **Package Provenance Definition and Purpose**
+
+In the npm CLI and package registry, package provenance comes in the form of software attestation that provides package consumers proof that a given package was:
+
+1. Published from the source repository it claims to be from  
+2. Using a specific build file executed on a GitHub hosted runner
+
+In GitHub’s [announcement](https://github.blog/security/supply-chain-security/introducing-npm-package-provenance/) their stated objective was to establish a root of trust for npm packages by:
+
+* Providing a direct line from the registry package to the source repository and commit it was published from  
+* Providing visibility into the build process and runner that generated the published package
+
+GitHub’s announcement referenced three attacks that they are seeking to prevent with package provenance: [malware in ua-parser-js](https://github.com/advisories/GHSA-pjwm-rvh2-c87w), [malware in coa](https://github.com/advisories/GHSA-73qr-pfmq-6rp8), [malware in rc](https://github.com/advisories/GHSA-g2q5-5433-rhrf). These attacks were all accomplished by malicious actors who:
+
+1. [ATT\&CK T1586.003](https://attack.mitre.org/techniques/T1586/003/): Compromised a maintainer’s npm account through undisclosed means  
+2. [CAPEC-538](https://capec.mitre.org/data/definitions/538.html): Added malware to the package and published the now malicious package to npm as an updated version of the legitimate package
```

Comment:
it doesn't actually prevent these attacks, though - it just means that attackers with repo access would have to be a bit more clever about obfuscating the source of the malicious code.